src/tree.rs

```diff
@@ -573,11 +573,20 @@ impl Tree {
               let target = pathdiff::diff_paths(&src_path, &base)
                 .ok_or_else(|| format_err!("failed to calculate path diff for {:?} -> {:?}", dst_path, src_path,))?;
 
+              let _ = std::fs::create_dir_all(&base)
+                .map_err(|err| eprintln!("warning: failed to create directory {:?}: {}", &base, err));
               let _ = create_symlink(target, &dst_path)
                 .map_err(|err| eprintln!("warning: failed to create symlink at {:?}: {}", dst_path, err));
             }
 
             FileOperation::CopyFile { .. } => {
+              let base = dst_path
```

Comment:
Can the base calculation & create_dir_all be extracted outside of the match statement, since they're common (except for the error messages)?

Reply:
Done